testutils/testutils-kmp/src/commonMain/kotlin/androidx/kruth/MapSubject.kt

```diff
@@ -35,4 +35,17 @@ class MapSubject<K, V> internal constructor(actual: Map<K, V>?) : Subject<Map<K,
             asserter.fail("Expected to contain $key, but was ${actual.keys}")
         }
     }
+
+    /** Fails if the map does not contain exactly the given set of entries in the given map. */
+    fun containsExactlyEntriesIn(expectedMap: Map<K, V>): Ordered {
+        requireNonNull(actual) { "Expected $expectedMap, but was null" }
+
+        if (expectedMap.isEmpty()) {
+            isEmpty()
+        } else if ((expectedMap.entries - actual.entries).isNotEmpty()) {
```

Comment:
This doesn't return `MapInOrder` and instead returns `NoopOrdered`, even if the entries are out of order.

For example this test should fail:
```
    @Test
    fun orderedMap() {
        val map = LinkedHashMap<Int, Int>()
        repeat(5) { map[it] = it }

        val unsortedMap = HashMap<Int, Int>()
        repeat(5) { unsortedMap[4 - it] = 4 - it }

        val reverseMap = LinkedHashMap<Int, Int>()
        repeat(5) { reverseMap[4 - it] = 4 - it }

        assertThat(map).containsExactlyEntriesIn(unsortedMap).inOrder()
        assertThat(map).containsExactlyEntriesIn(reverseMap).inOrder()
    }
```

Reply:
Oof, I'd just copied all the `containsExactlyEntriesIn` tests from Truth and assumed it'd cover all cases. Just updated the PR to throw when the elements are out of order.

Reply:
I believe they do have some, but it also depends on a different API that we don't have yet, which is probably why it got missed.

In the meantime, can we at least add the test I showed which fails without the `MapInOrder` change? Otherwise, everything else looks good to me !